### DISCUSSION TITLE
required not working on first step

### DISCUSSION THREAD

**Discussion by kasper-mol** [vueform/vueform#144] [Questions] [answered]
Hi,

i noticed something strange with the required rule on elements within steps. If for example i have a number input on my first form step i can continue to the next step even when the field is null. But if i then i go back to the first step and try to go to the second step the required validation seems to work as it should. 

I made this quickly using the builder to showcase it:
```vue
<template>
  <Vueform>
    <template #empty>
      <FormSteps>
        <FormStep
          name="page0"
          label="Step 1"
          :elements="[
            'number',
          ]"
        />
        <FormStep
          name="page1"
          label="Step 2"
          :elements="[
            'email',
          ]"
        />
      </FormSteps>

      <FormElements>
        <TextElement
          name="number"
          input-type="number"
          :rules="[
            'required',
            'numeric',
          ]"
          autocomplete="off"
          label="Number"
        />
        <TextElement
          name="email"
          input-type="email"
          :rules="[
            'nullable',
            'email',
          ]"
          label="Email"
        />
      </FormElements>

      <FormStepsControls />
    </template>
  </Vueform>
</template>
``` 

Any suggestions on a fix?

Cheers

**Comment by adamberecz** [comment, marked as answer]
I can't reproduce it - it throws the validation error on the first try for me. Maybe try to update to the latest version? It might be something already resolved.

**Reply by kasper-mol** [reply]
Sorry my bad, it had to do with our weird setup. It turns out the navigation to the next form via the `@finish` was triggered without checking for possible errors. Hence the errors were not shown because our second (and final) step is conditional to the value of the first. We were not aware that the `@finish` gets triggered even when there are validation errors in the form elements in the current step and the second is not activated due to the step conditions.
